### PR TITLE
qt 6.4.1

### PR DESCRIPTION
**Pull request by paperchalice**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
- Enable vulkan support.
- Using licenses listed in [`LICENSES`](https://github.com/qt/qtbase/tree/dev/LICENSES)(except 3rd party libraries)

Currently `qt` use bundled `harfbuzz` on macOS, if use brewed `harfbuzz`, it may report `On OS X, AAT is supported only with -qt-harfbuzz.`, but it is an obsolete warning and is [resolved](https://github.com/qt/qtbase/commit/6903cab48a4fbe1f0a1213261149b220718f2157) by upstream, should we use brewed version here?

### CODE REVIEW

File: Formula/qt-libiodbc.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
 class QtLibiodbc < Formula
   desc "Qt SQL Database Driver"
   homepage "https://www.qt.io/"
-  url "https://download.qt.io/official_releases/qt/6.4/6.4.0/submodules/qtbase-everywhere-src-6.4.0.tar.xz"
-  sha256 "cb6475a0bd8567c49f7ffbb072a05516ee6671171bed55db75b22b94ead9b37d"
-  license all_of: ["GPL-2.0-only", "GPL-3.0-only", "LGPL-2.1-only", "LGPL-3.0-only"]
```
Comment (chenrui333) [inline Formula/qt-libiodbc.rb:6]: Why do you update it to `any_of`?

Comment (chenrui333) [reply Formula/qt-libiodbc.rb:6]: But that is just that one file, right?

Comment (chenrui333) [reply Formula/qt-libiodbc.rb:6]: okay 👍

**Comment by chenrui333** [comment]
qt 6.4.1 test failure across board

**Comment by chenrui333** [comment]
[This commit](https://github.com/Homebrew/homebrew-core/pull/115926/commits/0888ee8de55006df54261b6392dfa5d4051bad68) should be separate out into two.

**Comment by cho-m** [comment]
Could be that your machine is using default dlopen lookup paths, which is more likely on Intel macOS due to standard `/usr/local` prefix.

Probably should check how Qt is searching for libvulkan. A workaround may be to set a `DYLD_*`/`LD_*` environment variable, but that would mean the feature is broken out-of-the-box.

**Comment by cho-m** [comment]
`[mvk-error] VK_ERROR_INCOMPATIBLE_DRIVER: Vulkan is not supported on this device. MoltenVK requires Metal, which is not available on this device.`

**Review by fxcoudert — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Comment by carlocab** [comment]
> Currently `qt` use bundled `harfbuzz` on macOS, if use brewed `harfbuzz`, it may report `On OS X, AAT is supported only with -qt-harfbuzz.`, but it is an obsolete warning and is [resolved](https://github.com/qt/qtbase/commit/6903cab48a4fbe1f0a1213261149b220718f2157) by upstream, should we use brewed version here?

When does the warning appear? Does upstream plan to get rid of it? This is fine for now, but using Homebrew `harfbuzz` instead of the bundled one should be something we look into for future updates.

**Comment by carlocab** [comment]
Does that mean the warning appears only during the build? If so, we can probably use `-system-harfbuzz`. If it appears during runtime then we may want to wait until upstream get rid of it.

**Comment by fxcoudert** [comment]
There was already no Linux bottle, since https://github.com/Homebrew/homebrew-core/pull/114917#issuecomment-1338146578

Now the issue is in the test. We get:

```
==> ./test
  qt.qpa.xcb: could not connect to display 
  qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
  This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
  
  Available platform plugins are: offscreen, minimalegl, linuxfb, eglfs, vnc, wayland-egl, vkkhrdisplay, xcb, wayland, minimal.
```

If you know how to fix this, a pull request would be welcome.

**Pull request closed without merging**